x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingDecisionAction.java

```diff
@@ -60,24 +65,37 @@ public XContentBuilder toXContent(final XContentBuilder builder, final Params pa
 
     public static class Response extends ActionResponse implements ToXContentObject {
 
-        public Response() {
+        private final SortedMap<String, AutoscalingDecision.MultipleAutoscalingDecision> decisions;
 
+        public Response(final SortedMap<String, AutoscalingDecision.MultipleAutoscalingDecision> decisions) {
+            this.decisions = Objects.requireNonNull(decisions);
         }
 
         public Response(final StreamInput in) throws IOException {
             super(in);
+            decisions = new TreeMap<>(in.readMap(StreamInput::readString, AutoscalingDecision.MultipleAutoscalingDecision::new));
         }
 
         @Override
-        public void writeTo(final StreamOutput out) {
-
+        public void writeTo(final StreamOutput out) throws IOException {
+            out.writeMap(decisions, StreamOutput::writeString, (o, decision) -> decision.writeTo(o));
         }
 
         @Override
         public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
             builder.startObject();
             {
-
+                builder.startArray("decisions");
+                {
+                    for (final Map.Entry<String, AutoscalingDecision.MultipleAutoscalingDecision> decision : decisions.entrySet()) {
+                        builder.startObject();
+                        {
+                            builder.field(decision.getKey(), decision.getValue());
```

Comment:
I think this should be just two fields "id" and "decision"? AFAICS, this will result in:
```
decisions : [
  {
     "key1" : { ... decision1 ... }
  },
  {
     "key2" : { ... decision2 ... }
  }
]

which seems to follow both the array and map pattern?

Reply:
Good point. I will address this in a follow-up when the framework is fleshed out a little bit more.